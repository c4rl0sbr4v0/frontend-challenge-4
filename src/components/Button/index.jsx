import React from 'react';
import PropTypes from 'prop-types';

/**
 *  Button: Render a Button
 */
const Button = ({ children, onClick, disabled }) => {
  return (
    <button className="button landbot-input-send" onClick={onClick} disabled={disabled}>
      {children}
    </button>
  );
};

export default Button;

Button.propTypes = {
  children: PropTypes.node,
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
};
Button.defaultProps = {
  children: '',
  onClick: null,
  disabled: false,
};
