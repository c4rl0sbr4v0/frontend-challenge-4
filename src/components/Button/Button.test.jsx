import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Button from '.';

test('Button Component', async () => {
  window.alert = jest.fn();
  const { container, getByText } = render(<Button onClick={() => alert('patata')}>Click me</Button>);
  expect(container.firstChild.classList.contains('landbot-input-send')).toBe(true);
  const button = getByText('Click me');
  fireEvent.click(button);
  expect(window.alert).toHaveBeenCalledTimes(1);
  expect(container).toMatchSnapshot();
});
