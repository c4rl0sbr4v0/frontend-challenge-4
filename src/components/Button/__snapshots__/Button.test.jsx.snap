// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component 1`] = `
<div>
  <button
    class="button landbot-input-send"
  >
    Click me
  </button>
</div>
`;
